shared/strbuf.c

```diff
@@ -55,12 +55,11 @@ char *strbuf_steal(struct strbuf *buf)
 {
 	char *bytes;
 
-	if (!buf_realloc(buf, buf->used + 1)) {
-		free(buf->bytes);
+	if (!buf_realloc(buf, buf->used + 1))
 		return NULL;
-	}
 
 	bytes = buf->bytes;
+	buf->bytes = NULL;
```

Comment:
The previous behaviour of "we failed to realloc, let's nuke everything" meant we had an UAF waiting to happen. I'm glad it's gone plus the extra NULL helps truly invalidate the strbuf.

Hell yeah \o/